doc/api/metrics.md

```diff
@@ -0,0 +1,1040 @@
+
+```js
+console.log(report.toDogStatsd()); // 'api.calls:1|c|service:web'
+```
+
+#### `metricReport.toGraphite()`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {string}
+
+Formats the metric report as a Graphite-compatible string.
+
+```js
+console.log(report.toGraphite()); // 'api.calls 1 1234567890'
+```
+
+#### `metricReport.toPrometheus()`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {string}
+
+Formats the metric report as a Prometheus-compatible string.
+
+```js
+console.log(report.toPrometheus()); // 'api_calls{service="web"} 1 1234567890.123'
+```
+
+### Class: `Metric`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+Manages the lifecycle of a metric channel and provides methods for reporting
+values to it. Each metric type holds a `Metric` instance which it reports to.
+
+#### `metric.type`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* {string}
+
+The type of the metric (e.g., 'counter', 'gauge', 'meter', 'periodicGauge',
+'timer', 'uniqueSet').
+
+#### `metric.name`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* {string}
+
+The name of the metric.
+
+#### `metric.meta`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* {Object}
+
+Additional metadata associated with the metric.
+
+#### `metric.channelName`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* {string}
+
+The name of the diagnostics_channel used for this metric.
+
+#### `metric.channel`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* {Channel}
+
+The diagnostics channel instance used for this metric.
+
+#### `metric.shouldReport`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* {boolean}
+
+Indicates whether the metric should report values. This can be used to
+conditionally enable or disable value preparation work.
+
+#### `metric.report(value[, meta])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `value` {number} The value to report.
+* `meta` {Object} Additional metadata for this report.
+* Returns: {metrics.MetricReport}
+
+Reports a value for the metric, creating a `MetricReport` instance.
+This bypasses the metric type specific methods, allowing direct reporting
+to a channel.
+
+Generally this method should not be used directly. Instead, use the
+specific methods provided by each metric type (e.g., `increment`, `reset`,
+`mark`, etc.) which internally call this method with the appropriate value and
+metadata.
+
+```mjs
+import { meter } from 'node:metrics';
+
+const apiCalls = meter('api.calls', { service: 'web' });
+
+apiCalls.metric.report(1); // Reports a value of 1
+apiCalls.metric.report(5, { endpoint: '/api/users' }); // Reports 5 with metadata
+```
+
+### Class: `Counter`
+
+* Extends: {metrics.Gauge}
+
+<!-- YAML
+added: REPLACEME
+-->
+
+A metric that only increases or decreases.
+
+#### `counter.metric`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* {metrics.Metric}
+
+The underlying metric instance used for reporting.
+
+#### `counter.increment([n[, meta]])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `n` {number} The amount to increment. **Default:** `1`
```

Comment:
Likewise, should `bigint` be supported? Won't mention these further... it applies to all of the new APIs here that accept or yield a `number` currently.